src/pg_client.py

```diff
@@ -18,13 +27,66 @@ def __str__(self) -> str:
         return str(self.value)
 
 
-def pg_engine(db_env: OrderbookEnv) -> engine:
-    """Returns a connection to postgres database"""
-    load_dotenv()
-    host = os.environ[f"{db_env}_ORDERBOOK_HOST"]
-    port = os.environ[f"{db_env}_ORDERBOOK_PORT"]
-    database = os.environ[f"{db_env}_ORDERBOOK_DB"]
-    user = os.environ[f"{db_env}_ORDERBOOK_USER"]
-    password = os.environ[f"{db_env}_ORDERBOOK_PASSWORD"]
-    db_string = f"postgresql+psycopg2://{user}:{password}@{host}:{port}/{database}"
-    return create_engine(db_string)
+@dataclass
+class DualEnvDataframe:
+    """
+    A pair of Dataframes primarily intended to be store query results
+    from production and staging orderbook databases
+    """
+
+    barn: DataFrame
+    prod: DataFrame
+
+    @staticmethod
+    def _pg_engine(db_env: OrderbookEnv) -> Engine:
+        """Returns a connection to postgres database"""
+        load_dotenv()
+        host = os.environ[f"{db_env}_ORDERBOOK_HOST"]
+        port = os.environ[f"{db_env}_ORDERBOOK_PORT"]
+        database = os.environ[f"{db_env}_ORDERBOOK_DB"]
+        user = os.environ[f"{db_env}_ORDERBOOK_USER"]
+        password = os.environ[f"{db_env}_ORDERBOOK_PASSWORD"]
+        db_string = f"postgresql+psycopg2://{user}:{password}@{host}:{port}/{database}"
+        return create_engine(db_string)
+
+    @classmethod
+    def from_query(cls, query: str) -> DualEnvDataframe:
+        """Fetch results of DB query on both prod and barn and returns the results as a pair"""
+        return cls(
+            barn=pd.read_sql(sql=query, con=cls._pg_engine(OrderbookEnv.PROD)),
+            prod=pd.read_sql(sql=query, con=cls._pg_engine(OrderbookEnv.BARN)),
+        )
+
+    def merge(self) -> DataFrame:
+        """Merges prod and barn dataframes via concatenation"""
+        # TODO - verify generic disjointness here.
+        return pd.concat([self.prod, self.barn])
+
+    @classmethod
+    def get_orderbook_rewards(cls, start_block: str, end_block: str) -> DataFrame:
+        """
+        Fetches and validates Orderbook Reward DataFrame as concatenation from Prod and Staging DB
+        """
+        cow_reward_query = (
+            open_query(query_file("orderbook/order_rewards.sql"))
+            .replace("{{start_block}}", start_block)
+            .replace("{{end_block}}", end_block)
+        )
+        dual_df = cls.from_query(cow_reward_query)
+
+        # Solvers do not appear in both environments!
+        # TODO - move this assertion into merge (and make it more general).
```

Comment:
Is there already an issue for these todos? Would be nice to link that here for future references.

Reply:
Issue: https://github.com/cowprotocol/solver-rewards/issues/125

Will also include in TODO comment.

Reply:
```suggestion
        # TODO - move this assertion into merge:
        #  https://github.com/cowprotocol/solver-rewards/issues/125
```

Reply:
Actually I have already proposed an answer in the issue. But we need to ship this so I won't implement it here.